tests/integration/security/rbac/grouplist/rbac_group_list_test.go

```diff
@@ -0,0 +1,110 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grouplist
+
+import (
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework/components/istio"
+
+	"istio.io/istio/tests/integration/security/rbac/util"
+
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/apps"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/util/connection"
+	"istio.io/istio/pkg/test/util/policy"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+const (
+	rbacClusterConfigTmpl  = "testdata/istio-clusterrbacconfig.yaml.tmpl"
+	rbacGroupListRulesTmpl = "testdata/istio-group-list-rbac-rules.yaml.tmpl"
+	groupsScopeJwt         = "eyJhbGciOiJSUzI1NiIsImtpZCI6IkRIRmJwb0lVcXJZOHQyenBBMnFYZkNtcj" +
+		"VWTzVaRXI0UnpIVV8tZW52dlEiLCJ0eXAiOiJKV1QifQ.eyJleHAiOjM1MzczOTExMDQsImdyb3VwcyI6WyJncm91cD" +
+		"EiLCJncm91cDIiXSwiaWF0IjoxNTM3MzkxMTA0LCJpc3MiOiJ0ZXN0aW5nQHNlY3VyZS5pc3Rpby5pbyIsInNjb3BlI" +
+		"jpbInNjb3BlMSIsInNjb3BlMiJdLCJzdWIiOiJ0ZXN0aW5nQHNlY3VyZS5pc3Rpby5pbyJ9.EdJnEZSH6X8hcyEii7c" +
+		"8H5lnhgjB5dwo07M5oheC8Xz8mOllyg--AHCFWHybM48reunF--oGaG6IXVngCEpVF0_P5DwsUoBgpPmK1JOaKN6_pe" +
+		"9sh0ZwTtdgK_RP01PuI7kUdbOTlkuUi2AO-qUyOm7Art2POzo36DLQlUXv8Ad7NBOqfQaKjE9ndaPWT7aexUsBHxmgi" +
+		"Gbz1SyLH879f7uHYPbPKlpHU6P9S-DaKnGLaEchnoKnov7ajhrEhGXAQRukhDPKUHO9L30oPIr5IJllEQfHYtt6IZvl" +
+		"NUGeLUcif3wpry1R5tBXRicx2sXMQ7LyuDremDbcNy_iE76Upg"
+	rbacTestRejectionCode = "401"
+)
+
+var (
+	inst          istio.Instance
+	isMtlsEnabled bool
+)
+
+func setupConfig(cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+	isMtlsEnabled = cfg.IsMtlsEnabled()
+	cfg.Values["sidecarInjectorWebhook.rewriteAppHTTPProbe"] = "true"
+}
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite("rbac_group_list", m).
+		RequireEnvironment(environment.Kube).
+		Setup(istio.SetupOnKube(&inst, setupConfig)).
+		Run()
+}
+
+func TestGroupListRBAC(t *testing.T) {
+	ctx := framework.NewContext(t)
+	defer ctx.Done(t)
+	ctx.RequireOrSkip(t, environment.Kube)
+
+	env := ctx.Environment().(*kube.Environment)
+	g := galley.NewOrFail(t, ctx, galley.Config{})
+	p := pilot.NewOrFail(t, ctx, pilot.Config{
+		Galley: g,
+	})
+	appInst := apps.NewOrFail(t, ctx, apps.Config{Pilot: p, Galley: g})
+
+	appA, _ := appInst.GetAppOrFail("a", t).(apps.KubeApp)
+	appB, _ := appInst.GetAppOrFail("b", t).(apps.KubeApp)
+	appC, _ := appInst.GetAppOrFail("c", t).(apps.KubeApp)
+
+	cases := []util.TestCase{
+		// Port 80 is where HTTP is served
+		{Request: connection.Connection{To: appB, From: appA, Port: 80, Protocol: apps.AppProtocolHTTP, Path: "/xyz"}, ExpectAllowed: false},
```

Comment:
Instead of testing request without JWT token, I think it's better to test request with valid JWT token but with `group != group1` or `request.auth.claims[scope] != scope1`. These requests should be rejected as it doesn't have the expected group/claims, but the JWT token validation is allowed.

A request without JWT token is rejected by authentication which has nothing to do with RBAC, we should avoid making the RBAC test relying on that. Also if you only test request with valid JWT token, the deny code will not change.

Reply:
Done: changed to use a valid JWT that does doesn't have the expected group/list claims.